tiledb/sm/compressors/delta_compressor.cc

```diff
@@ -201,12 +210,11 @@ void Delta::decompress(
       break;
     case Datatype::FLOAT32:
     case Datatype::FLOAT64:
-      throw StatusException(
-          Status_CompressionError("Cannot decompress tile with Delta; Float "
-                                  "datatypes are not supported"));
+      throw DeltaCompressorException(
+          "Decompression is not yet supported for float datatypes.");
     default:
-      throw StatusException(Status_CompressionError(
-          "Cannot compress tile with Delta; Unsupported datatype"));
+      throw DeltaCompressorException(
+          "Decompression failed; Unsupported datatype");
```

Comment:
Nit: Use `datatype_str` to say which datatype is unsupported.

Reply:
`datatype_str` will return an empty string here because each `Datatype` case is covered above.